src/runtime_src/core/tools/common/Table2D.cpp

```diff
@@ -0,0 +1,187 @@
+/**
+ * Copyright (C) 2022 Xilinx, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+// ------ I N C L U D E   F I L E S -------------------------------------------
+// Local - Include Files
+#include "Table2D.h"
+#include "core/common/system.h"
+
+// 3rd Party Library - Include Files
+#include <algorithm>
+#include <boost/format.hpp>
+
+const std::string no_data_string = "N/A";
+
+Table2D::Table2D() : m_entry_count(0)
+{
+}
+
+Table2D::Table2D(const std::vector<HeaderData>& headers) : m_entry_count(0)
+{
+    addHeaders(headers);
+}
+
+void
+Table2D::addHeader(const HeaderData& header)
+{
+    // Format the header into a table data struct
+    ColumnData column;
+    column.header = header;
+    // Populate the header with the appropriate amount of N/A
+    column.entry_slices = std::vector<std::string>();
+    for (size_t entry_count = 0; entry_count < m_entry_count; entry_count++)
+        column.entry_slices.push_back(no_data_string);
+    m_table.push_back(column);
+}
+
+void
+Table2D::addHeaders(const std::vector<HeaderData>& headers)
+{
+    for (auto& header : headers)
+        addHeader(header);
+}
+
+bool
+Table2D::removeHeader(const std::string& header)
+{
+    // Search through each clumn looking for a matching header
+    for (size_t column_index = 0; column_index < m_table.size(); ++column_index) {
+        // If a matching header is found remove the entire column
+        if (m_table[column_index].header.name.compare(header) == 0) {
+            m_table.erase(m_table.begin() + column_index);
+            return true;
+        }
+    }
+    // Return false when no match is found
+    return false;
+}
+
+size_t
+Table2D::addEntry(const std::vector<std::string>& entry)
+{
+    // Keep track of each new entry
+    ++m_entry_count;
```

Comment:
Do we really need a separate variable for the number of entries.  Can't we simple determine it via the m_table collection.

Reply:
Removed